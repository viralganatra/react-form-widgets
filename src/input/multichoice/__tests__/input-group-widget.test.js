import React from 'react';
import InputGroupWidget from '../input-group-widget';

const Item = (props) => <input {...props} />;

function noop() {}

const props = {
    items: [
        { label: 'test 1', value: '1' },
        { label: 'test 2', value: '2' },
    ],
    value: ['2'],
    isChecked: () => true,
    onChange: noop,
};

describe('InputGroupWidget', () => {
    let Component;

    beforeAll(() => {
        Component = InputGroupWidget(Item);
    });

    it('should render a list of labels with inputs, based on the `items` prop', () => {
        const wrapper = shallow(
            <Component {...props} />,
        );

        expect(wrapper).toMatchSnapshot();
        expect(() => InputGroupWidget({})).toThrowErrorMatchingSnapshot();
    });

    it('should validate the prop types', () => {
        const wrapperInvalidItems = () => shallow(
            <Component {...props} items={2} />,
        );

        const wrapperInvalidIsChecked = () => shallow(
            <Component {...props} isChecked={[]} />,
        );

        const wrapperInvalidOnChange = () => shallow(
            <Component {...props} onChange={[]} />,
        );

        expect(wrapperInvalidItems).toThrowErrorMatchingSnapshot();
        expect(wrapperInvalidIsChecked).toThrowErrorMatchingSnapshot();
        expect(wrapperInvalidOnChange).toThrowErrorMatchingSnapshot();
    });
});
