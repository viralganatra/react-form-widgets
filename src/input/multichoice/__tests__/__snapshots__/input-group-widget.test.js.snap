exports[`InputGroupWidget should render a list of labels with inputs, based on the \`items\` prop 1`] = `
<div>
  <Item
    checked={true}
    label={
      Object {
        "content": "test 1",
      }
    }
    onChange={[Function]}
    value="1" />
  <Item
    checked={true}
    label={
      Object {
        "content": "test 2",
      }
    }
    onChange={[Function]}
    value="2" />
</div>
`;

exports[`InputGroupWidget should render a list of labels with inputs, based on the \`items\` prop 2`] = `
"Expected \"WrappedComponent\" provided as the first argument to
        CheckboxContainerHOC to be a function. Instead, received
        object."
`;

exports[`InputGroupWidget should validate the prop types 1`] = `
"Warning: Failed prop type: Invalid prop \`items\` of type \`number\` supplied to \`InputMultichoiceGroup\`, expected an array.
    in InputMultichoiceGroup"
`;

exports[`InputGroupWidget should validate the prop types 2`] = `
"Warning: Failed prop type: Invalid prop \`isChecked\` of type \`array\` supplied to \`InputMultichoiceGroup\`, expected \`function\`.
    in InputMultichoiceGroup"
`;

exports[`InputGroupWidget should validate the prop types 3`] = `
"Warning: Failed prop type: Invalid prop \`onChange\` of type \`array\` supplied to \`InputMultichoiceGroup\`, expected \`function\`.
    in InputMultichoiceGroup"
`;
